### DISCUSSION TITLE
Server Side Rendering Time

### DISCUSSION THREAD

**Discussion by nikhilatcompro** [nuxt/nuxt#28354] [Questions]
Is there a way to determine the time taken by the server to render a page/route?

Imagine I have a page in my nuxt 3 application that calls a single API to fetch some data and then converts the Vue js component into HTML which is to be send to the client.

how can i determine the time taken to create this HTML?

**Comment by sirgeon** [comment]
Yes theres a way. You could create a middleware to measure the time taken.

```ts
export default defineEventHandler((event) => {
    const start = Date.now();

    // hook to run when the response is about to be sent
    event.node.res.on('finish', () => {
        const end = Date.now();
        const duration = end - start;
        console.log(`Request to ${event.req.url} took ${duration}ms to render.`);
    });
});
```

and then register in nuxt config
```ts
export default defineNuxtConfig({
    nitro: {
        prerender: {
            routes: ['/']
        },
    },
    routeRules: {
        '/**': { middlewares: ['urMiddleWarename'] },
    }
});

```

hope i could help u
